### PR TITLE
Update CI Continuous Integration page and add CircleCi example [skip ci]

### PR DESCRIPTION
**Pull request by michaelbadcrumble**
Just an update to CI Continuous Integration page because I think it would help people can install both Meson and Ninja using `pip install meson ninja`, this should make CI YAML files easier to read.

CircleCi example is added to provide a new option for Meson build projects when choosing a CI provider.

### CODE REVIEW

File: docs/markdown/Continuous-Integration.md [hunk 1]
```diff
@@ -44,6 +44,65 @@ script:
...
+  meson_ubuntu_build:
+    executor: meson_ubuntu_builder
+    steps:
+      - checkout
+      - run: meson setup builddir --backend ninja
+      - run: ninja -C builddir
+      - run: meson test -C builddir
+
+  meson_debain_build:
+    executor: meson_debain_builder
```
Comment (scivision) [inline docs/markdown/Continuous-Integration.md]:
```yml
executor: meson_debian_builder
```

File: docs/markdown/Continuous-Integration.md [hunk 2]
```diff
@@ -67,17 +126,14 @@ platform:
...
 install:
-  # Download ninja
-  - cmd: mkdir C:\ninja-build
-  - ps: (new-object net.webclient).DownloadFile('https://github.com/mesonbuild/cidata/raw/master/ninja.exe', 'C:\ninja-build\ninja.exe')
   # Set paths to dependencies (based on architecture)
   - cmd: if %arch%==x86 (set PYTHON_ROOT=C:\python37) else (set PYTHON_ROOT=C:\python37-x64)
   # Print out dependency paths
   - cmd: echo Using Python at %PYTHON_ROOT%
   # Add necessary paths to PATH variable
   - cmd: set PATH=%cd%;C:\ninja-build;%PYTHON_ROOT%;%PYTHON_ROOT%\Scripts;%PATH%
```
Comment (scivision) [inline docs/markdown/Continuous-Integration.md]:
```yml
- cmd: set PATH=%cd%;%PYTHON_ROOT%;%PYTHON_ROOT%\Scripts;%PATH%
```

File: docs/markdown/Continuous-Integration.md [hunk 3]
```diff
@@ -44,6 +44,65 @@ script:
...
+
+```yaml
+version: 2.1
+
+executors:
+  # Your dependencies would go in the docker images that represent
+  # the Linux distributions you are supporting
+  meson_ubuntu_builder:
+    docker:
+      - image: michaelbadcruble/ubuntu-sys
```
Comment (scivision) [inline docs/markdown/Continuous-Integration.md]: maybe the image should be generic so readers realize they need to choose one like
```yml
    docker:
      - image: your_dockerhub_username/ubuntu-sys
```

**Comment by scivision** [comment]
#6252

File: docs/markdown/Continuous-Integration.md (hunk 1 above)
Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: This calls one of the executors.  In this case the Ubuntu one is being called.

File: docs/markdown/Continuous-Integration.md (hunk 3 above)
Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: Sure thing.  Update applied.

File: docs/markdown/Continuous-Integration.md (hunk 2 above)
Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: Removing unused Ninja value from the set path command

File: docs/markdown/Continuous-Integration.md (hunk 1 above)
Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: No No Sorry Debian.

File: docs/markdown/Continuous-Integration.md [hunk 4]
```diff
@@ -44,6 +44,65 @@ script:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then SDKROOT=$(xcodebuild -version -sdk macosx Path) meson builddir && ninja -C builddir test; fi
 ```
 
+## CircleCi for Linux (with Docker)
+
+For CI on Linux, [CircleCi](https://circleci.com/) is probably
+your best bet. Here's a sample `yml` file for use with that.
```
Comment (textshell) [inline docs/markdown/Continuous-Integration.md]: I don't think this is true. I'm sure CircleCi works, but so does any other of the common CI providers that offer linux.

Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: This most definitely works, please check out my dummy/templates for your self.

[D](https://github.com/michaelbadcrumble/d-project)

Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: However I think you probably meant the words being marked up in green.  So I would be happy to rewrite that as soon as possible.

Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: Done.

File: docs/markdown/Continuous-Integration.md (hunk 1 above)
Comment (scivision) [reply docs/markdown/Continuous-Integration.md]: debain => debian

File: docs/markdown/Continuous-Integration.md (hunk 3 above)
Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: Ok @scivision does that mean I should revert the Travis example for you to work on?

Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: Ok reverted the Travis example so you can work on it.

File: docs/markdown/Continuous-Integration.md (hunk 4 above)
Comment (textshell) [reply docs/markdown/Continuous-Integration.md]: Thanks for changing the wording. But i still feel "fantastic" is not what the official meson documentation should say about a particular commercial CI company without a solid foundation for such a wording. I'm happy to have a CircleCi example, but i don't think the meson documentation should be using such wording here.

File: docs/markdown/Continuous-Integration.md (hunk 3 above)
Comment (scivision) [reply docs/markdown/Continuous-Integration.md]: no need to revert, I also want to add Github Actions so go ahead and I'll do it later.

**Comment by scivision** [comment]
this should be squashed before merging

File: docs/markdown/Continuous-Integration.md (hunk 4 above)
Comment (jpakkane) [reply docs/markdown/Continuous-Integration.md]: I agree. Comments about external service providers should be neutral.

Comment (jpakkane) [reply docs/markdown/Continuous-Integration.md]: It should go more along the lines of "if you wish to use CircleCI, then <rest of the text>".

Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: Ok made some changes and hopefully it works for the documentation.

Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: However as a suggestion I think there should be a guideline for how the documentation should be written.

**Comment by jpakkane** [comment]
Thanks.

File: docs/markdown/Continuous-Integration.md (hunk 4 above)
Comment (michaelbadcrumble) [reply docs/markdown/Continuous-Integration.md]: 😕 not sure what happened but I know it could not have been the documentation change.

**Comment by michaelbadcrumble** [comment]
Your welcome

File: docs/markdown/Continuous-Integration.md (hunk 4 above)
Comment (textshell) [reply docs/markdown/Continuous-Integration.md]: I'd say use your best guess and don't worry too much when that does not line up with what others expect in the end and you have to reword something.

I think review of actual documentation works a lot better than anyone trying to come up with a complicated guide.

**Pull request merged**